### PR TITLE
set hostap crypto configuration to default - WIFI_NM_WPA_SUPPLICANT_CRYPTO_ALT, remove PSA dependency and fix build errors 

### PR DESCRIPTION
**Pull request by VivekUppunda**
1) Set the default crypto selected as WIFI_NM_WPA_SUPPLICANT_CRYPTO_ALT.
2) Do not select PSA as default and let it be a choice to the user to select
3) Fix build errors in Kconfig seen
    1) mbedtls_ssl_conf_dhm_min_bitlen - requires MBEDTLS_DHM_C and MBEDTLS_SSL_CLI_C
         mbedtls_ssl_conf_dh_param_ctx - requires MBEDTLS_DHM_C and MBEDTLS_SSL_SRV_C
         the configurations MBEDTLS_SSL_CLI_C and MBEDTLS_SSL_SRV_C are internal definitions which are enabled 
         when CONFIG_MBEDTLS_TLS_VERSION_1_2 is enabled
    2) `wpa_eapol_key_mic` function `hmac_md5 call` which requires MBEDTLS_MD5

### CODE REVIEW

File: modules/hostap/Kconfig [hunk 1]
```diff
@@ -105,11 +105,11 @@ config WIFI_NM_WPA_SUPPLICANT_WEP
 
 choice WIFI_NM_WPA_SUPPLICANT_CRYPTO_BACKEND
 	prompt "WPA supplicant crypto implementation"
-	default WIFI_NM_WPA_SUPPLICANT_CRYPTO
+	default WIFI_NM_WPA_SUPPLICANT_CRYPTO_ALT
 	help
 	  Select the crypto implementation to use for WPA supplicant.
-	  WIFI_NM_WPA_SUPPLICANT_CRYPTO_ALT support enterprise
-	  and DPP. And use Mbedtls PSA apis for HW acceleration.
+	  WIFI_NM_WPA_SUPPLICANT_CRYPTO_ALT supports enterprise mode
```
Comment (krish2718) [inline modules/hostap/Kconfig:111]: Please add the rationale for this in the commit log and also missing full stop.

File: modules/hostap/Kconfig [hunk 2]
```diff
@@ -164,7 +164,6 @@ endchoice
 
 config WIFI_NM_WPA_SUPPLICANT_CRYPTO_MBEDTLS_PSA
 	bool "Crypto Platform Secure Architecture support for WiFi"
-	default y if WIFI_NM_WPA_SUPPLICANT_CRYPTO_ALT
```
Comment (krish2718) [inline modules/hostap/Kconfig:167]: @fengming-ye hope this is fine, not every implementation supports PSA, so, this should be explicitly enabled.

Comment (fengming-ye) [reply modules/hostap/Kconfig:167]: yes, make sense

**Comment by VivekUppunda** [comment]
@krish2718 , @fengming-ye - can you please ack the PR if it looks good please?

**Review by fengming-ye — APPROVED** [review]

**Review by jukkar — APPROVED** [review]

**Review by krish2718 — APPROVED** [review]

**Pull request merged**